pkg/templates/danglingservicemonitor/template.go

```diff
@@ -0,0 +1,64 @@
+package danglingservicemonitor
+
+import (
+	"fmt"
+
+	k8sMonitoring "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/danglingservicemonitor/internal/params"
+	v1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Dangling Service Monitor",
+		Key:         "dangling-servicemonitor",
+		Description: "Flag service monitors which do not match any service",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.ServiceMonitor},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(_ params.Params) (check.Func, error) {
+			return func(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				serviceMonitor, ok := object.K8sObject.(*k8sMonitoring.ServiceMonitor)
+				if !ok {
+					return nil
+				}
+				labelSelectors := serviceMonitor.Spec.Selector.MatchLabels
+				if len(labelSelectors) == 0 {
+					return []diagnostic.Diagnostic{{
+						Message: "service monitor has no selector specified",
+					}}
+				}
+				labelSelector, err := metaV1.LabelSelectorAsSelector(&metaV1.LabelSelector{MatchLabels: serviceMonitor.Spec.Selector.MatchLabels})
+				if err != nil {
+					return []diagnostic.Diagnostic{{
+						Message: fmt.Sprintf("service monitor has invalid label selector: %v", err),
```

Comment:
could you add a unit test for this

Reply:
added in dd3a7d59f58ac66efdd4ccab061060d467634cc6